design/authenticators/authn-k8s/inject_client_cert_solution_design.md

```diff
@@ -0,0 +1,195 @@
+| An authn-client container with permissions on the "ssl" directory | An "inject-client-cert" request is sent | - The response code is 200 OK<br />- The cert is in the container |                                                              |
+| An authn-client container without permissions on the "ssl" directory | An "inject-client-cert" request is sent | - The response code is 401<br />- The error is written to the log |                                                              |
+| A service-account that has no sufficient permissions         | An "inject-client-cert" request is sent | - The response code is 401<br />- The error is written to the log | This test is different than the one above because the error type is different. This request will fail today as well while the previous one would have succeeded (although he cert wouldn't be installed). |
+
+## Logs
+[//]: # "Fill in the table below to depict the log messages that can enhance the supportability of your solution"
+[//]: # "You can use this tool to generate a table - https://www.tablesgenerator.com/markdown_tables#"
+
+| **Scenario** | **Log message** |
+|--------------|-----------------|
+| Timeout reached for injection kubectl exec command | Command timed out after {0} seconds in container '{1}' of pod '{2}' |
+| Verify cert injection (if we choose to do so)      | Verifying SSL certificate is present in {0-container-name}:{1-cert-file-path}                |
+| Before executing a command using kubectl exec      | Executing command '{0-command}' in container {1-container-name} of pod {2-pod-name} |
+
+### Audit 
+[//]: # "Does this solution require additional audit messages?"
+There are no new audit messages in this fix. We already write the inject-client-cert request to the audit log.
+
+## Documentation
+[//]: # "Add notes on what should be documented in this solution. Elaborate on where this should be documented. If the change is in open-source projects, we may need to update the docs in github too. If it's in Conjur and/or DAP mention which products are affected by it"
+This is an internal process so it doesn't require any new documentation.
+
+## Open questions
+[//]: # "Add any question that is still open. It makes it easier for the reader to have the open questions accumulated here istead of them being acattered along the doc"
+1. What happens if we run an "ls" command on a file that is still being populated with tar? 
+
+## Delivery plan
+[//]: # "Break the solution into tasks"
+
+### Total
+[//]: # "Write the total estimation"
+
+EE: 22 days
+
+### Parallelism
+
+[//]: # "Suggest how many team members can work on the solution in parallel"
+
+- 2 people can work on this solution but it's ok to have only 1.
+
+### Designs 
+[//]: # "Note and estimate the tasks which are related to designs which needed for the solution"
+
+- Add a diagram to the solution design once we decide on the solution
+  - EE: 1 day
+
+### Spikes and researches
+
```

Comment:
Do you plan to spike all the options? Isn't there a preffered option that worth testing first?

Reply:
i ordered the spikes and added the comment `The spikes are ordered by priority. If a spike assesses that the desired solution is feasible then we will not perform the following spikes. `